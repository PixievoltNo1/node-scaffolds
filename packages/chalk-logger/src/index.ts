import type { Mutable } from '@guanghechen/utility-types'
import { partOptionsFromArgs } from './commander'
import type { Level } from './level'
import type { ILoggerOptions } from './logger'
import { Logger } from './logger'

export * from './color'
export * from './commander'
export * from './level'
export * from './logger'

export class ChalkLogger extends Logger {
  /**
   * prefix of logger.name
   */
  protected basename: string | null = null
  /**
   * name passed into .setName()
   */
  protected divisionName: string | null = null

  constructor(options?: ILoggerOptions, args?: string[]) {
    const optionsFromArgs = partOptionsFromArgs(args || [])
    const resolvedOptions = {
      ...options,
      ...optionsFromArgs,
      flags: { ...options?.flags, ...optionsFromArgs.flags },
    }

    super(resolvedOptions)
    const basename = resolvedOptions?.name ?? ''
    this.setBaseName(basename)
  }

  /**
   * update logger's level
   * @param level
   */
  public setLevel(level: Level | null | undefined): void {
    if (level == null) return
    const self = this as Mutable<this>
    self.level = level
  }

  /**
   * update logger's name
   * @param name
   */
  public setName(name: string | null): void {
    const resolvedName: string = [this.basename, name]
      .filter((x): x is string => x != null && x.length > 0)
      .join(' ')
    const self = this as Mutable<this>
    self.name = resolvedName
    this.divisionName = name
  }

  /**
   * update basename of logger
   * @param basename
   */
  public setBaseName(basename: string | null): void {
    this.basename = basename
    this.setName(this.divisionName)
  }

  /**
   * update logger's mode
   * @param mode
   */
  public setMode(mode: 'normal' | 'loose'): void {
    const self = this as Mutable<this>
    self.mode = mode
  }
}

export default ChalkLogger
