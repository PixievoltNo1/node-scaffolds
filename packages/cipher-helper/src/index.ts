export * from './_base'
export * from './aes'
export * from './catalog'
export * from './types/catalog'
export * from './types/cipher-helper'
export * from './util/buffer'
export * from './util/key'
