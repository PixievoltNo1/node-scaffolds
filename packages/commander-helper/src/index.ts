export * from './command'
export * from './fs'
export * from './git'
export * from './merge'
export * from './option'
export * from './yarn'
