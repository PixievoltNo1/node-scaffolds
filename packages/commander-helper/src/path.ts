import path from 'path'

/**
 * Calc absolute path of p under the workspace
 *
 * @param workspace
 * @param targetPath
 */
export function absoluteOfWorkspace(
  workspace: string,
  targetPath?: string | null,
): string {
  if (targetPath == null) return workspace
  const absoluteDir: string = path.resolve(workspace, targetPath)
  return absoluteDir
}

/**
 * Calc relative path to workspace
 *
 * @param workspace
 * @param targetDir
 */
export function relativeOfWorkspace(
  workspace: string,
  targetDir: string,
): string {
  const absoluteDir = absoluteOfWorkspace(workspace, targetDir)
  const relativeDir: string = path.relative(workspace, absoluteDir)
  return path.normalize(relativeDir)
}
