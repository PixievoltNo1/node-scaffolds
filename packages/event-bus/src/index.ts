export * from './SimpleEventBus'
export * from './types'
