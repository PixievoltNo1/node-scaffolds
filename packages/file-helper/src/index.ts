export * from './big-file-helper'
export * from './types'
export * from './util/fs'
export * from './util/split-file'
export * from './util/stream'
