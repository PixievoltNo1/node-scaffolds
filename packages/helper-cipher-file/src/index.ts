export * from './types/IFileCipher'
export * from './types/IFileCipherBatcher'
export * from './types/IFileCipherFactory'
export * from './util/fingerprint'

export * from './FileCipher'
export * from './FileCipherBatcher'
export * from './FileCipherCatalog'
export * from './FileCipherCatalog.readonly'
export * from './FileCipherFactory'
