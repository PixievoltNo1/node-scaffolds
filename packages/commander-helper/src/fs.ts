import { isFileSync } from '@guanghechen/file-helper'
import fs from 'fs-extra'
import yaml from 'js-yaml'
import path from 'path'

/**
 * Load configuration file with format .json / .yml / .yaml
 *
 * @param filepath
 * @param logger
 */
export async function loadJsonOrYaml(
  filepath: string,
  encoding = 'utf-8',
): Promise<unknown | never> {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): Promise<string> =>
    fs.readFile(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = await loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = await loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}

/**
 * Load configuration file with format .json / .yml / .yaml  (synchronizing)
 *
 * @param filepath
 * @param logger
 */
export function loadJsonOrYamlSync(
  filepath: string,
  encoding = 'utf-8',
): unknown | never {
  if (!isFileSync(filepath))
    throw new Error(`${filepath} is an invalid file path`)

  const loadContent = (): string =>
    fs.readFileSync(filepath, encoding as BufferEncoding)

  let result: unknown
  const extname = path.extname(filepath)
  switch (extname) {
    case '.json':
      {
        const content: string = loadContent()
        result = JSON.parse(content)
      }
      break
    case '.yml':
    case '.yaml':
      {
        const content: string = loadContent()
        result = yaml.load(content, { filename: filepath, json: true })
      }
      break
    default:
      throw new Error(
        `Only files in .json / .yml / .ymal format are supported. filepath(${filepath}`,
      )
  }
  return result
}
