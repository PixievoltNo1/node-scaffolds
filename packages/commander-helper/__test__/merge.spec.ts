import { defaultMergeStrategies, merge } from '../src'

describe('defaultMergeStrategies', function () {
  const prevOptions: any = {
    a: {
      b: {
        c: 'txt',
      },
    },
    b: 'dfe',
    c: ['x', 'y'],
  }

  test('replace', function () {
    expect(
      defaultMergeStrategies.replace(prevOptions, {
        a: {
          b: {
            d: 'waw',
          },
        },
        d: 'dfe',
      }),
    ).toEqual({
      a: {
        b: {
          d: 'waw',
        },
      },
      d: 'dfe',
    })

    expect(defaultMergeStrategies.replace(prevOptions, null)).toEqual(
      prevOptions,
    )

    expect(defaultMergeStrategies.replace(prevOptions, undefined)).toEqual(
      prevOptions,
    )
  })

  test('retain', function () {
    expect(
      defaultMergeStrategies.retain(prevOptions, {
        a: {
          b: {
            d: 'waw',
          },
        },
        d: 'dfe',
      }),
    ).toEqual(prevOptions)

    expect(defaultMergeStrategies.retain(prevOptions, null)).toEqual(
      prevOptions,
    )

    expect(defaultMergeStrategies.retain(prevOptions, undefined)).toEqual(
      prevOptions,
    )
  })

  test('concat', function () {
    expect(defaultMergeStrategies.concat(prevOptions, { a: 1 } as any)).toEqual(
      prevOptions,
    )

    expect(defaultMergeStrategies.concat(['a', 'b'], ['c'])).toEqual([
      'a',
      'b',
      'c',
    ])

    expect(defaultMergeStrategies.concat(['a', 'b'], [])).toEqual(['a', 'b'])

    expect(defaultMergeStrategies.concat([], ['a', 'b'])).toEqual(['a', 'b'])
  })

  test('assign', function () {
    expect(
      defaultMergeStrategies.assign(prevOptions, {
        a: {
          b: {
            d: 'waw',
          },
        },
        d: 'dfe',
      }),
    ).toEqual({
      a: {
        b: {
          d: 'waw',
        },
      },
      b: 'dfe',
      c: ['x', 'y'],
      d: 'dfe',
    })

    expect(defaultMergeStrategies.assign(prevOptions, null)).toEqual({
      a: {
        b: {
          c: 'txt',
        },
      },
      b: 'dfe',
      c: ['x', 'y'],
    })
  })
})

describe('merge', function () {
  const options: any[] = [
    {
      a: {
        b: {
          c: 'txt',
        },
      },
      b: 'dfe',
      c: ['x', 'y'],
      d: 'd',
    },
    {
      a: {
        b: {
          d: 'waw',
        },
      },
      c: ['a', 'b'],
    },
  ]

  test('replace', function () {
    expect(
      merge(options, {
        a: defaultMergeStrategies.assign,
        b: defaultMergeStrategies.retain,
        c: defaultMergeStrategies.concat,
      }),
    ).toMatchSnapshot()

    expect(merge(options)).toMatchSnapshot()
  })
})
