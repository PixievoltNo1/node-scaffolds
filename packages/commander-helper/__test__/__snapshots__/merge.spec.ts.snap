// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge replace 1`] = `
Object {
  "a": Object {
    "b": Object {
      "d": "waw",
    },
  },
  "b": "dfe",
  "c": Array [
    "x",
    "y",
    "a",
    "b",
  ],
  "d": "d",
}
`;

exports[`merge replace 2`] = `
Object {
  "a": Object {
    "b": Object {
      "d": "waw",
    },
  },
  "b": "dfe",
  "c": Array [
    "a",
    "b",
  ],
  "d": "d",
}
`;
